src/promptflow-tools/tests/test_handle_openai_error.py

```diff
@@ -105,51 +74,46 @@ def test_aoai_invalid_request_error_with_bad_deployment(self, aoai_provider):
             "The API deployment for this resource does not exist. If you created the deployment "
             "within the last 5 minutes, please wait a moment and try again."
         )
-        error_msg = to_openai_error_message(InvalidRequestError(message=raw_message, param=None))
-        error_codes = "UserError/OpenAIError/InvalidRequestError"
+        error_codes = "UserError/OpenAIError/NotFoundError"
         with pytest.raises(WrappedOpenAIError) as exc_info:
             aoai_provider.chat(prompt=f"user:\n{prompt_template}", deployment_name=deployment)
-        assert error_msg == exc_info.value.message
+        assert raw_message in exc_info.value.message
         assert exc_info.value.error_codes == error_codes.split("/")
 
     def test_rate_limit_error_insufficient_quota(self, azure_open_ai_connection, mocker: MockerFixture):
-        dummyEx = RateLimitError("Something went wrong", json_body={"error": {"type": "insufficient_quota"}})
-        mock_method = mocker.patch("promptflow.tools.aoai.openai.Completion.create", side_effect=dummyEx)
+        dummyEx = RateLimitError("Something went wrong", response=httpx.Response(
+            429, request=httpx.Request('GET', 'https://www.example.com')), body={"type": "insufficient_quota"})
+        mock_method = mocker.patch("openai.resources.Completions.create", side_effect=dummyEx)
         error_codes = "UserError/OpenAIError/RateLimitError"
         with pytest.raises(WrappedOpenAIError) as exc_info:
             completion(connection=azure_open_ai_connection, prompt="hello", deployment_name="text-ada-001")
         assert to_openai_error_message(dummyEx) == exc_info.value.message
         assert mock_method.call_count == 1
         assert exc_info.value.error_codes == error_codes.split("/")
 
-    def test_non_retriable_connection_error(self, azure_open_ai_connection, mocker: MockerFixture):
-        dummyEx = APIConnectionError("Something went wrong")
-        mock_method = mocker.patch("promptflow.tools.aoai.openai.Completion.create", side_effect=dummyEx)
-        error_codes = "UserError/OpenAIError/APIConnectionError"
-        with pytest.raises(WrappedOpenAIError) as exc_info:
-            completion(connection=azure_open_ai_connection, prompt="hello", deployment_name="text-ada-001")
-        assert to_openai_error_message(dummyEx) == exc_info.value.message
-        assert mock_method.call_count == 1
-        assert exc_info.value.error_codes == error_codes.split("/")
-
     @pytest.mark.parametrize(
         "dummyExceptionList",
         [
             (
                 [
-                    RateLimitError("Something went wrong"),
-                    ServiceUnavailableError("Something went wrong"),
```

Comment:
APIStatusError, but 503

Reply:
Add InternalServerError("Something went wrong", response=httpx.Response(503, request=httpx.Request('GET', 'https://www.example.com')), body=None) to test ServiceUnavailableError. OpenAI will throw InternalServerError with status code >=500